### PR TITLE
Starting to make 1.0-compatible

### PR DESCRIPTION
**Pull request by matbesancon**

### CODE REVIEW

**Comment by matbesancon** [comment]
@blegat current issue is:
```julia
(ConvexHull) pkg> test ConvexHull
   Testing ConvexHull
 Resolving package versions...
ERROR: LoadError: LoadError: UndefVarError: arraytype not defined
```

`Polyhedra.arraytype` seems to be removed, how has it been replaced?

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,13 @@ os:
     - linux
     - osx
 julia:
-    - 0.6
+    - 0.7
```
Comment (blegat) [inline .travis.yml]: You can drop it and require at least Julia v1.0 as Polyhedra v0.5 requires Julia v1.0

**Comment by blegat** [comment]
> `Polyhedra.arraytype` seems to be removed, how has it been replaced?

`Polyhedra.vectortype`

**Comment by matbesancon** [comment]
the tests are now breaking "only" in the tests. Problem is, these tests are calling the Polyhedra library tests. I noticed CDDLib doesn't do that anymore. Should this be changed?

**Comment by matbesancon** [comment]
Fixes #12 
(soon)

**Comment by blegat** [comment]
> I noticed CDDLib doesn't do that anymore.

What do you mean ? CDDLib still runs the Polyhedra tests

**Comment by matbesancon** [comment]
oops yup I had missed it

**Comment by blegat** [comment]
It seems `RecipeBase` need to be added for tests: https://travis-ci.org/JuliaPolyhedra/ConvexHull.jl/jobs/521296399#L541

**Comment by matbesancon** [comment]
just fixed

**Comment by matbesancon** [comment]
There is a mistake that I can't explain now, an assertion on `!== nothing` in Polyhedra itself

File: test/REQUIRE [hunk 2]
```diff
@@ -1,3 +1,2 @@
 JuMP
 GLPK
```
Comment (blegat) [inline test/REQUIRE]: Let's remove this file and drop Julia < v1.0

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+authors = ["Joey Huchette"]
+version = "0.1.0"
+
+[deps]
+Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
+GenericLinearAlgebra = "14197337-ba66-59df-a3e3-ca00e7dcff7a"
+Polyhedra = "67491407-f73d-577b-9b50-8179a7c68029"
+Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
+RowEchelon = "af85af4c-bcd5-5d23-b03a-a909639aa875"
+
```
Comment (blegat) [inline Project.toml]: Add a `[compat]`  with `julia = "1"` and `Polyhedra = "0.5"`

**Comment by blegat** [comment]
> There is a mistake that I can't explain now, an assertion on !== nothing in Polyhedra itself

Could you give the full trace ? The tests seems to be passing on Travis

File: Project.toml (hunk 3 above)
Comment (matbesancon) [reply Project.toml]:
```
ERROR: empty intersection between Polyhedra@0.4.5 and project compatibility 0.5.0-0.5
```

**Comment by matbesancon** [comment]
```
ERROR: LoadError: LoadError: LoadError: AssertionError: lp_solver !== nothing
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:326 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] include(::String) at ./client.jl:403
 [6] top-level scope at none:0
 [7] include at ./boot.jl:326 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1038
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] include(::String) at ./client.jl:403
 [11] top-level scope at none:0
 [12] include at ./boot.jl:326 [inlined]
 [13] include_relative(::Module, ::String) at ./loading.jl:1038
 [14] include(::Module, ::String) at ./sysimg.jl:29
 [15] include(::String) at ./client.jl:403
 [16] top-level scope at none:0
in expression starting at /home/mbesancon/.julia/packages/Polyhedra/Su6Kf/test/solvers.jl:48
in expression starting at /home/mbesancon/.julia/dev/ConvexHull/test/polyhedron.jl:6
in expression starting at /home/mbesancon/.julia/dev/ConvexHull/test/runtests.jl:10
ERROR: Package ConvexHull errored during testing
```

**Comment by blegat** [comment]
This means that no LP solver is installed.
Try installing GLPK

File: Project.toml (hunk 3 above)
Comment (blegat) [reply Project.toml]: Did you do `] add Polyhedra@0.4.5` in the past ? In this case, you can do `] free Polyhedra` or do `] activate .` in the `dev/ConvexHull` folder.

Comment (matbesancon) [reply Project.toml]: ok nevermind here, solved

**Comment by matbesancon** [comment]
@blegat CI is passing in a very ugly way, the errors seem to come from Polyhedra?

**Comment by blegat** [comment]
It seems to be missing a method like
https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/2c205dd19f48ccd75292efce012933bd26917b1d/src/defaultlibrary.jl#L57

**Comment by matbesancon** [comment]
I think I found it, there were multiple inner constructors for the ConvexHull.ConvexHullPolyhedron type which is not allowed

**Comment by matbesancon** [comment]
Ok I fixed this, now the tests are erroring because some methods are missing :/

**Comment by matbesancon** [comment]
@blegat could you take a look at this error?
https://travis-ci.org/JuliaPolyhedra/ConvexHull.jl/jobs/535375155#L546
I'm not sure which methods are still required there

**Comment by blegat** [comment]
I am looking into it.

> I'm not sure which methods are still required there

This is quite tricky, I usually look at what I did for `Polyhedra/src/defaultpolyhedron.jl` and get inspiration from it ^^

**Comment by blegat** [comment]
Try merging the `up10` branch of this repo into yours

**Comment by matbesancon** [comment]
It's better now, there are still problems with the errors hard-coded from Polyhedra:
https://travis-ci.org/JuliaPolyhedra/ConvexHull.jl/jobs/547705458#L546

**Comment by blegat** [comment]
That's because removing redundancy requires doing double description atm, this should fallback to solving LPs in the future which would resolve the issue. You can exclude the test for now.

**Comment by matbesancon** [comment]
looks good, not sure coverage is really comparable between Julia versions

**Pull request merged**